src/drivers/Kconfig

```diff
@@ -5,6 +5,112 @@ config DMIC
 	depends on CAVS
 	default n
 	help
-	  Select for DMIC driver
+		Select this to enable DMIC driver. The DMIC driver provides
+		as DAI the SoC direct attach digital microphones interface.
+		.
+		.
+
+if DMIC
+
+choice
+	prompt "FIR decimation coefficients set"
+	default DMIC_FIR_FULL
+
+config DMIC_FIR_FULL
+	bool "Full set"
+	select DMIC_FIR_DECIMATE_BY_2
+	select DMIC_FIR_DECIMATE_BY_3
+	select DMIC_FIR_DECIMATE_BY_4
+	select DMIC_FIR_DECIMATE_BY_5
+	select DMIC_FIR_DECIMATE_BY_6
+	select DMIC_FIR_DECIMATE_BY_8
+	help
+		This option adds to previous all currently defined FIR
+		coefficients sets to support sample rates 8 - 96 kHz with
+		several microphone clock rates. The tables increase the size
+		of the driver so this option should not be used in minimal
+		systems.
+
+config DMIC_FIR_LOW_MEMORY
+	bool "Small set"
+	select DMIC_FIR_DECIMATE_BY_2
+	select DMIC_FIR_DECIMATE_BY_6
+	help
+		This option is used to minimize driver footprint but
+		preserve support for 48 kHz and 16 kHz sample rates
+		at typical 2.4 MHz microphone clock rate. The option
+		enables decimation factors 2 and 6.
+
+config DMIC_FIR_CUSTOM
+	bool "Custom set"
+	help
+		This option is used to select each supported decimation
+		factor.
+		.
+		.
+
+endchoice
+
+menu "Decimation factors"
+	visible if DMIC_FIR_CUSTOM
+
+config DMIC_FIR_DECIMATE_BY_2
```

Comment:
Ditto here for CAVS prefix

Reply:
Yep, this is Intel and CAVS specific. I'll update the PR.